warehouse/manage/views.py

```diff
@@ -593,12 +656,42 @@ def create_macaroon(self):
 
         response = {**self.default_response}
         if form.validate():
+            macaroon_caveats = {"permissions": form.validated_scope, "version": 1}
             serialized_macaroon, macaroon = self.macaroon_service.create_macaroon(
                 location=self.request.domain,
                 user_id=self.request.user.id,
                 description=form.description.data,
-                caveats={"permissions": form.validated_scope, "version": 1},
+                caveats=macaroon_caveats,
             )
+            self.user_service.record_event(
+                self.request.user.id,
+                tag="account:api_token:added",
+                ip_address=self.request.remote_addr,
+                additional={
+                    "description": form.description.data,
+                    "caveats": macaroon_caveats,
+                },
+            )
+            if "projects" in form.validated_scope:
```

Comment:
I'm wondering if we also want to expose when a user creates an API token that has the same effective permissions, in other words a "user scoped" token. It seems like it might not be the _best_ idea, but it's also probably not relevant to this PR at this time.

Reply:
Yep, that was the topic of this NB: https://github.com/pypa/warehouse/pull/6339#issuecomment-520879774

One minor issue with recording project events on a "user scoped" token's creation is that only preexisting projects will receive the creation event, while future projects will silently allow the token but not contain the corresponding event. That's not necessarily worse than the current PR behavior, just something to keep in mind. We could also generally mitigate the problem by listing associated API tokens separately somewhere in the project management view.

Reply:
In the interest of getting this PR merged, I've opened a new ticket to follow up on this: https://github.com/pypa/warehouse/issues/6441